data-prepper-plugins/armeria-common/src/main/java/org/opensearch/dataprepper/GrpcRequestExceptionHandler.java

```diff
@@ -60,6 +63,9 @@ private Status handleExceptions(final Throwable e) {
         } else if (e instanceof BadRequestException) {
             badRequestsCounter.increment();
             return createStatus(e, Status.INVALID_ARGUMENT);
+        } else if ((e instanceof StatusRuntimeException) && (message.contains("Invalid protobuf byte sequence") || message.contains("Can't decode compressed frame"))) {
```

Comment:
I couldn't think of any other way because getCause() is null

Reply:
Also, grpc getStatus() on grpc StatusRuntimeException is just showing the error as INTERNAL. I do not see any other option here.

Reply:
@kkondaka , I think this can work for now. Which project is creating this exception - Armeria or gRPC? I think we should reach out to them and let them know that there is not enough information to handle this error as we'd like without relying on user error messages, which can be brittle.

Reply:
Note, that I am migrating the GrpcRequestExceptionHandler to implement `GoogleGrpcExceptionHandlerFunction`. This also changes this method slightly. But it will most likely not help with this problem: https://github.com/opensearch-project/data-prepper/pull/4294/files